src/Schema.ts

```diff
@@ -137,7 +173,64 @@ class Schema {
 		}, new Set<number>());
 
 		return [...positions];
-	};
+	}
+
+	/** Build the dictionary path map for additional dictionaries provided as schema options */
+	private buildDictionaryPaths(
+		dictionaries: Record<string, DictionaryDefinition>,
+	): Map<string, DictionaryTypeDetail> {
+		// Add reference for _id --> @ID by default
+		const dictPaths = new Map<string, DictionaryTypeDetail>([
+			['_id', { dictionary: '@ID', dataTransformer: new StringDataTransformer() }],
+		]);
+
+		return Object.entries(dictionaries).reduce((acc, [queryProperty, dictionaryDefinition]) => {
+			if (typeof dictionaryDefinition === 'string') {
+				return acc.set(queryProperty, {
+					dictionary: dictionaryDefinition,
+					dataTransformer: new StringDataTransformer(),
+				});
+			}
+
+			const { type, dictionary } = dictionaryDefinition;
+
+			switch (type) {
+				case 'string':
+					return acc.set(queryProperty, {
+						dictionary,
+						dataTransformer: new StringDataTransformer(),
+					});
+				case 'number':
+					return acc.set(queryProperty, {
+						dictionary,
+						dataTransformer: new NumberDataTransformer(),
```

Comment:
Can't you specify an oconv value in a virtual dictionary?

Reply:
Maybe I'm not quite following the question, but the number query transformer doesn't do anything other than return the string representation of whatever value was provided, just like the string query transformer.  The expectation is that the query engine would handle the decimals.

Reply:
@kthompson23 Not quite sure if there's an issue that you're concerned about or not.  Let me know if there's something.

Reply:
OK that makes sense.